comp/core/agenttelemetry/impl/sender.go

```diff
@@ -306,29 +337,30 @@ func (s *senderImpl) flushSession(ss *senderSession) error {
 		return err
 	}
 
-	// Send the payload. In the future we want to move this functionality/code into DefaultForwarder
-	// because it provides retry, caching, URL management, API key rotation at runtime, flush to disk,
-	// backoff logic etc.
-	req, err := http.NewRequest("POST", s.endpointURL, bytes.NewReader(reqBody))
-	if err != nil {
-		return err
-	}
-	s.addHeaders(req, payload.RequestType, s.cfgComp.GetString("api_key"), strconv.Itoa(len(reqBody)))
-	resp, err := s.client.Do(req.WithContext(ss.cancelCtx))
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if resp != nil && resp.Body != nil {
-			resp.Body.Close()
+	// Send the payload to all endpoints
+	for _, ep := range s.endpoints.Endpoints {
+		url := buildURL(ep)
+		req, err := http.NewRequest("POST", url, bytes.NewReader(reqBody))
+		if err != nil {
+			return err
 		}
-	}()
+		s.addHeaders(req, payload.RequestType, ep.GetAPIKey(), strconv.Itoa(len(reqBody)))
+		resp, err := s.client.Do(req.WithContext(ss.cancelCtx))
+		if err != nil {
+			return err
```

Comment:
If this (or `NewRequest`) fails for any reason, we will break out of the loop and not try any additional endpoints. 
Is this intended? I expect we want to send to the other endpoints if the first one fails.

Reply:
Good catch. I have addressed it in the last commit. Thank you.

Reply:
Have you push? 😄 

The last commit seems to be a merge commit

Reply:
wow, indeed, but I cannot see it anymore even locally. Oh boy looking, thank you

Reply:
I had problem with merging though. I did it locally but in UI it was "behind" and so I merge it from UI, will see if it is still in history, damn github

Reply:
found it, I have committed to a wrong branch. Github was good, I was bad

Reply:
Looks like I lost it :( Addressing it. Thank you for the catch. Because there were a commit labeled with code review description I presume that it was the one I just did but it was older commit

Reply:
all set, sorry about that